node/src/p2p.rs

```diff
@@ -66,37 +44,100 @@ pub enum P2pError {
     WorkerDied,
 }
 
-type Result<T, E = P2pError> = std::result::Result<T, E>;
+#[derive(Debug)]
+pub struct P2p {
+    cmd_tx: flume::Sender<P2pCmd>,
+}
 
-struct Worker {
-    swarm: Swarm<Behaviour>,
-    header_sub_topic_hash: TopicHash,
-    cmd_rx: flume::Receiver<P2pCmd>,
+pub struct P2pArgs {
+    pub transport: Boxed<(PeerId, StreamMuxerBox)>,
+    pub network_id: String,
+    pub local_keypair: Keypair,
+    pub bootstrap_peers: Vec<Multiaddr>,
+    pub listen_on: Vec<Multiaddr>,
 }
 
 #[derive(Debug)]
-enum P2pCmd {
+pub enum P2pCmd {
```

Comment:
I think this shouldn't be public because we will implement wrappers for the commands

Reply:
making it private would mean making everything from `Service` trait private too and preventing any external implementations

Reply:
That is true, please use `#[doc(hide)]` for now. If we have an issue in the future we may need to wrap it in a struct and have the field as private.

Reply:
hidden